### PR TITLE
Add YAML definition for code-related semantic attributes.

### PR DESCRIPTION
**Pull request by vovencij**
## Changes
As requested in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1628#discussion_r523246765 this PR adds [source code related span attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/span-general.md#source-code-attributes) to the list of semantic attributes.

### CODE REVIEW

**Comment by Oberon00** [comment]
Hi! Thank you for adding YAML for this. You now need to replace the table in https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/span-general.md#source-code-attributes with the HTML comments to have it auto-generated and run `make table-generation`.

**Comment by arminru** [comment]
Hi @vovencij! Thanks for your contribution.
Please reference the semantic convention you added in `span-general.md` and have the table automatically generated as described [here](https://github.com/open-telemetry/opentelemetry-specification/blob/master/semantic_conventions/README.md):
```
<!-- semconv code-->
... table will be generated here ...
<!-- endsemconv -->
```
This way, it is verified that the YAML syntax is fine and we can compare the generated table with the current one.

File: specification/performance-benchmark.md [hunk 1]
```diff
@@ -1,70 +1,70 @@
...
-Measure dynamic memory consumption, e.g. heap, for the same scenario as above
-CPU Usage section with 15 seconds duration.
-
-## Report
-
-### Report Format
-
-All the numbers above should be measured multiple times (suggest 10 times at
-least) and reported.
+# Performance Benchmark of OpenTelemetry API
```
Comment (Oberon00) [inline specification/performance-benchmark.md:1]: Please revert this file.

**Comment by vovencij** [comment]
After running `make table-generation` `specification/performance-benchmark.md` was regenerated. Should I roll it back or leave as is?

File: specification/trace/semantic_conventions/span-general.md [hunk 2]
```diff
@@ -168,9 +168,11 @@ For an HTTP server span, this would be the function that handles the incoming re
...
 about the span.
 
+<!-- semconv code -->
 | Attribute  | Type | Description  | Example  | Required |
 |---|---|---|---|---|
-| `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name). | `serveRequest` | No |
+| `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name) | `serverRequest` | No |
 | `code.namespace` | string | The "namespace" within which `code.function` is defined. Usually the qualified class or module name, such that `code.namespace` + some separator + `code.function` form a unique identifier for the code unit. | `com.example.MyHttpService` | No |
-| `code.filepath` | string | The source code file name that identifies the code unit as uniquely as possible (preferably an absolute file path). | `/usr/local/MyApplication/src/main/java/com/example/MyHttpService.java` | No |
+| `code.filepath` | string | The source code file name that identifies the code unit as uniquely as possible (preferably an absolute file path). | `/usr/local/MyApplication/content_root/app/index.php` | No |
```
Comment (Oberon00) [inline specification/trace/semantic_conventions/span-general.md:176]: Why this change?

File: semantic_conventions/trace/general.yaml [hunk 3]
```diff
@@ -94,4 +94,30 @@ groups:
...
+        examples: 'read:message, write:files'
+  - id: code
+    prefix: code
+    brief: >
+      These attributes allow to report this unit of code and therefore to provide more context about the span.
+    attributes:
+      - id: function
+        type: string
+        brief: >
+          The method or function name, or equivalent (usually rightmost part of the code unit's name)
```
Comment (Oberon00) [inline semantic_conventions/trace/general.yaml:106]:
```suggestion
          The method or function name, or equivalent (usually rightmost part of the code unit's name).
```

File: specification/performance-benchmark.md (hunk 1 above)
Comment (Oberon00) [reply specification/performance-benchmark.md:1]: > After running make table-generation specification/performance-benchmark.md was regenerated

That seems very odd. I don't think the tool touches files that do not contain these tables. Otherwise we might have a line ending problem. Maybe the spec is missing .gitattributes and the file was commited with CRLF?

Comment (Oberon00) [reply specification/performance-benchmark.md:1]: I will create an issue to check the line endings. Please still revert this file in the meantime. Sorry for the trouble!

File: specification/trace/semantic_conventions/span-general.md (hunk 2 above)
Comment (vovencij) [reply specification/trace/semantic_conventions/span-general.md:176]: I find that PHP is a better example to demonstrate the `filepath` attribute. How Java apps are deployed there is usually no information about the location of the source file. In runtime there are only compiled classes. So in case of Java this attribute is not relevant. It will contain the location of the file on developer's or build machine at best. Whereas PHP and languages alike usually have the source code actually deployed in the Apache/nginx server and this attribute makes all the sense in the world.

File: specification/performance-benchmark.md (hunk 1 above)
Comment (vovencij) [reply specification/performance-benchmark.md]: Reverted.

**Review by iNikem — APPROVED** [review]

**Review by Oberon00 — APPROVED** [review]
Thank you!

File: semantic_conventions/trace/general.yaml [hunk 4]
```diff
@@ -94,4 +94,30 @@ groups:
...
+  - id: code
+    prefix: code
+    brief: >
+      These attributes allow to report this unit of code and therefore to provide more context about the span.
+    attributes:
+      - id: function
+        type: string
+        brief: >
+          The method or function name, or equivalent (usually rightmost part of the code unit's name).
+        examples: serverRequest
```
Comment (arminru) [inline semantic_conventions/trace/general.yaml:107]:
```suggestion
        examples: serveRequest
```

I think _serving a request_ makes more sense for a class called `MyHttpService` rather than having a "server request".

File: specification/trace/semantic_conventions/span-general.md [hunk 5]
```diff
@@ -168,9 +168,11 @@ For an HTTP server span, this would be the function that handles the incoming re
 The attributes listed below allow to report this unit of code and therefore to provide more context
 about the span.
 
+<!-- semconv code -->
 | Attribute  | Type | Description  | Example  | Required |
 |---|---|---|---|---|
-| `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name). | `serveRequest` | No |
+| `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name). | `serverRequest` | No |
```
Comment (arminru) [inline specification/trace/semantic_conventions/span-general.md:174]:
```suggestion
| `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name). | `serveRequest` | No |
```

**Review by arminru — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Comment by iNikem** [comment]
@open-telemetry/technical-committee Can this be merged?

**Comment by Oberon00** [comment]
We usually consider YAML changes editorial, but this has only approvers with write access from a single company (not that I really have a say here, but: @iNikem would you be interested in becoming an specs-traces-approver or specs-approver?).
Is the change urgent?

**Comment by iNikem** [comment]
@Oberon00 if you find my opinions useful, then I certainly can contribute as specs-traces-approver :)

**Comment by iNikem** [comment]
> Is the change urgent?

As it seems that we have a general consensus on this, the actual merge is not urgent. I just don't like WIP :)

**Comment by arminru** [comment]
It's an editorial change except for slightly changing one example but that should be uncontroversial.
@vovencij could you please rebase onto the latest master? The checks currently shown as pending were moved to Github actions in the meantime. Once the checks pass, we can merge it.
